e2e/nx-aws-cache-e2e/tests/aws-cache.encrypted.test.ts

```diff
@@ -0,0 +1,93 @@
+import { AwsCache } from '../../../packages/nx-aws-cache/src/tasks-runner/aws-cache';
+import { MessageReporter } from '../../../packages/nx-aws-cache/src/tasks-runner/message-reporter';
+import { Logger } from '../../../packages/nx-aws-cache/src/tasks-runner/logger';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { createWriteStream } from 'fs';
+import { Decrypt, EncryptConfig } from '../../../packages/nx-aws-cache/src/tasks-runner/encryptor';
+import { pipeline, Readable } from 'stream';
+import { promisify } from 'util';
+
+describe('Test aws put and get unencrypted file', () => {
+  let awsCache: AwsCache;
+  let hash = new Date().getTime().toString();
+  let cacheDirectory = path.join(os.tmpdir(), `aws-cache`);
+  let cacheDirectorySave = path.join(os.tmpdir(), `aws-cache-decompress`);
+  let fileContent = 'console.log(123)';
+  let filePath = '';
+
+  const config = {
+    encryptionFileKey: 'Pbfk58EpcK7IxTxWwSXNsTAKmzhJQE+99vkpGftyJg8=',
+    awsAccessKeyId: 'minio',
+    awsSecretAccessKey: 'minio123',
+    awsBucket: 'test',
+    awsEndpoint: 'http://127.0.0.1:9000',
+    awsForcePathStyle: true,
+    awsRegion: 'us-east-1',
+  };
+
+  beforeEach(() => {
+    awsCache = new AwsCache(config, new MessageReporter(new Logger()));
+
+    fs.mkdirSync(cacheDirectory, {
+      recursive: true,
+    });
+    fs.mkdirSync(cacheDirectorySave, {
+      recursive: true,
+    });
+    const fileDir = path.join(cacheDirectory, `${hash}/outputs`);
+    fs.mkdirSync(fileDir, { recursive: true });
+    filePath = path.join(fileDir, `test.js`);
+    fs.writeFileSync(filePath, fileContent);
+  });
+
+  it('Should be defined', () => {
+    expect(awsCache).toBeDefined();
+  });
+
+  it('Should save encrypted data in s3 file, and read an unencrypted', async () => {
+    await awsCache.store(hash, cacheDirectory);
+    await awsCache.retrieve(hash, cacheDirectorySave);
+    const extractedFilePath = path.join(cacheDirectorySave, `${hash}/outputs/test.js`);
+    expect(fs.existsSync(extractedFilePath)).toBeTruthy();
+    expect(fs.readFileSync(extractedFilePath).toString()).toBe(fileContent);
+  });
+
+  it('Check that file is encrypted in s3 at rest', async () => {
+    await awsCache.store(hash, cacheDirectory);
+
+    // Check that file is encrypted in s3
+    const s3 = new S3Client({
```

Comment:
Here, you are using local minio for testing s3 logic, but this is not ok especially because it makes it hard to run on CI (local run is ok). It's better to just mock s3 (e.g. using `aws-sdk-client-mock`.

Reply:
Mock in e2e tests? I will not make e2e tests with mock

Reply:
I will not use mock for this. I know from experience that this will cause problems in the future, I prefer to do an integration test with full write process to normal s3 server.

Reply:
I prefer test with docker-compose for full write process. I dont know why it's hard to run in CI, in gitlab without docker would be hard but in github? I will change to aws-sdk-client-mock

Reply:
I prefer test with docker-compose for full write process. I dont know why it's hard to run in CI, in gitlab without docker would be hard but in github? I will change to aws-sdk-client-mock

Reply:
Changed to aws-sdk-client-mock but I have to create mock like this:

```js
// Aws-sdk-client-mock not mock CreateMultipartUploadCommand and UploadPartCommand for @aws-sdk/lib-storage I dont know why. 
/// There is a issue with this: https://github.com/m-radzikowski/aws-sdk-client-mock/issues/118
    s3Mock.onAnyCommand((input: { Body?: Buffer }) => {
      if (input?.Body) {
        uploadedData = input.Body;
        return { ETag: '1' };
      }
      /* eslint consistent-return: "warn", no-useless-return: "warn" */
      return;
    });
```